medallion/test/config/test_config_loading.py

```diff
@@ -0,0 +1,353 @@
+
+    We do this since we don't want to attempt to create those files for parsing
+    in any of our actual functional tests below.
+    """
+    assert m_cfg.load_config.__defaults__ == (
+        m_cfg.DEFAULT_CONFFILE, m_cfg.DEFAULT_CONFDIR,
+    )
+
+
+@mock.patch.dict("os.environ", clear=True)
+def test_load_config_default_file_missing(subtests, tmp_path):
+    """
+    Confirm that nothing bad happens if the default config file is missing.
+    """
+    with mock.patch(
+        "medallion.config.DEFAULT_CONFFILE", str(tmp_path / "medallion.conf"),
+    ) as mock_conf_file_default:
+        with subtests.test(msg="conf_dir omitted"):
+            config = m_cfg.load_config(conf_file=mock_conf_file_default)
+            assert config == {}
+        with subtests.test(msg="conf_dir is None"):
+            config = m_cfg.load_config(
+                conf_file=mock_conf_file_default, conf_dir=None,
+            )
+            assert config == {}
+
+
+def test_load_config_non_default_file_missing(subtests, tmp_path):
+    """
+    Confirm an exception is raised if a non-default config file is missing.
+    """
+    conf_file = tmp_path / "missing.conf"
+    exc_patt = re.escape(repr(str(conf_file)))
+    with pytest.raises(FileNotFoundError, match=exc_patt):
+        m_cfg.load_config(conf_file=conf_file)
+
+
+@mock.patch.dict("os.environ", clear=True)
+def test_load_config_default_dir_missing(subtests, tmp_path):
+    """
+    Confirm that nothing bad happens if the default config dir is missing.
+    """
+    with mock.patch(
+        "medallion.config.DEFAULT_CONFDIR", str(tmp_path / "medallion.d"),
+    ) as mock_conf_dir_default:
+        with subtests.test(msg="conf_file omitted"):
+            config = m_cfg.load_config(conf_dir=mock_conf_dir_default,)
+            assert config == {}
+        with subtests.test(msg="conf_file is None"):
+            config = m_cfg.load_config(
+                conf_dir=mock_conf_dir_default, conf_file=None,
+            )
+            assert config == {}
+
+
+def test_load_config_non_default_dir_missing(tmp_path):
+    """
+    Confirm an exception is raised if a non-default config dir is missing.
+    """
+    conf_dir = tmp_path / "missing.d"
+    exc_patt = re.escape(repr(str(conf_dir)))
+    with pytest.raises(FileNotFoundError, match=exc_patt):
+        m_cfg.load_config(conf_dir=conf_dir)
+
+
+@mock.patch.dict("os.environ", clear=True)
+def test_load_config_logging(subtests, tmp_path, caplog):
+    """
+    Ensure that config loading emits log messages about paths and data.
+    """
+    confdata = {"foo": "bar"}
+    conf_file = tmp_path / "medallion.conf"
+    json.dump(confdata, conf_file.open("w"))
+    caplog.set_level(logging.DEBUG, logger="medallion.config")
+    with subtests.test(msg="Used as config file"):
+        assert m_cfg.load_config(conf_file=conf_file) == confdata
+        assert f"load configuration from {str(conf_file)!r}" in caplog.text
```

Comment:
This line causes the tests to fail on my Windows machine. The conf_file path has 4 slashes in between directories, while caplog is printing with 2 slashes. I'd imagine doing an extra `re.escape` might work, or whatever the opposite is.

Reply:
I think this is actually to do with the repr which is being used to format the `conf_file`. That's only being done to add single quotes so I can just include them in the format string myself. I just sanity checked with this, so I am pretty sure my updated diff will play nice on your Windows box :)

```python
>>> backslash = chr(92)
>>> print(backslash)
\
>>> print(repr(backslash))
'\\'
>>> print(f"{backslash!r}")
'\\'
>>> print(f"--{backslash!r}--")
--'\\'--
```